object_database/web/cells/util.py

```diff
@@ -312,3 +312,23 @@ def MarginSides(amount, cell=None):
         return CellDecorator(lambda cell: MarginSides(amount, cell), f"MarginSides({amount})")
 
     return CustomInset(cell, "margin", top=0, right=amount, bottom=0, left=amount)
+
+
+def BackgroundColor(color, cell=None):
```

Comment:
I was surprised to see a function that is made to look like a constructor, but that modifies the cell object, rather than creating a new object with the necessary changes. I see that this is the style of this util.py file, and there might be a reason for it, just found it surprising.

Reply:
Yes it is surprising. This is the interface that was originally requested when we started making these modifiers. I think the idea was that it is much more clear when declaring Cell structures. We are bordering on a DSL here at this point.

Reply:
I think I wanted that interface when we would have thought of this as a 'new cell'. Maybe a better way to express this would be `BackgroundColor("red") * cell`, and then add a `clone` method to all Cell objects (it can be a shallow copy, not a deep copy) that these cell modifiers can use to change the cell behavior.  That would get rid of the part of this pattern that is the most confusing, namely 'modifying' the cell object.

Thoughts?

Reply:
Why would there need to be a clone at all? We could just have the `*` operator add additional properties to the existing instance (in this case to style exportData), no?

Reply:
So the idea behind having a 'clone' operation is that when you have expression-tree-like syntax, you want the objects flowing through them to be 'value-like' (not mutable). When I write "A*(B+C)" I usually expect that I'm producing a new value, not modifying anything.  The pattern of 'A*B' modifying 'B' in some way is bad because then you can't write `(A1*B + A2*B)` and expect it to work.  This is not just a pedantic detail - it's totally reasonable to make a function that takes a cell and then uses that cell twice with different formatting.

So I guess I like the pattern, but would like to also make it so that `exportData` also doesn't get directly modified, but instead gets cloned and updated.